### ISSUE TITLE
GTNH Huge memory bloat on fresh install. exitcode -1073740791

### ISSUE THREAD

**Issue by BaneWilliams** [GTNewHorizons/GT-New-Horizons-Modpack#14818]
### Your GTNH Discord Username

banewilliams

### Your Pack Version

2.4.0

### Your Server

SP

### Java Version

Java 20

### Type of Server

Single Player

### Your Actions

JAVA 21!

On launch, the game loaded fine. Then I closed it down at the main menu, and it wouldn't load back up and crashed with the error above. I thought I had recreated it by changing the RAM size in Prism but it wasn't reproducable. 

Then it loaded fine, and I was playing it on stream - it crashed and generated the following error output:

`OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000007f7800000, 8388608, 0) failed; error='The paging file is too small for this operation to complete' (DOS error/errno=1455)
#
There is insufficient memory for the Java Runtime Environment to continue.
Native memory allocation (mmap) failed to map 8388608 bytes for G1 virtual space
An error report file with more information is saved as:
C:\Users\Bane's PC\AppData\Roaming\PrismLauncher\instances\GT_New_Horizons_2.4.0_Java_17-20.minecraft\hs_err_pid13608.log
[1425.057s][warning][os] Loading hsdis library failed
Process crashed with exitcode -1073740791."`

I have a mid-high end computer, with 32GB of RAM and AMD CPU/GPU I have 1TB free on my boot drive, NVMe SSD.

This was a fresh minecraft install, installed for this specific purpose. I have not played minecraft in about 5 years, and never on this PC before. This is a fresh worldgen, I've played 4 hours and have never played the pack before.

The only things I changed in Prism was increasing the default Java RAM allocation to 4gb and adding J21 to the system via the installation link in your installation guide.

My computer, while streaming, commits about 14gb of RAM with chrome + discord and a few background processes. But I noticed during play that my system was committing huge amounts of RAM, way more than the game should realistically need, and this was leading to general instability (see through chunks, disappearing fluids) 

![image](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/80081089/3cf16239-42fc-44c2-a3ec-8b762971e7e9)
My usual system amount.

![image](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/80081089/49174f07-18bd-4138-95e9-4f74ee15d6d0)
While loading GTNH

![image](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/80081089/ee7fe247-f2cc-41dd-8b08-277a1afc1d55)
Main Menu GTNH

Me realising that it's not an instant jump up to massive RAM usage, but it is over a few minutes of very basic gameplay. https://youtu.be/Nk4zT41tQ_4

I tried using J19 just in case it was a Java version issue.

Going to do a few things to see if I can figure out the specific cause from my hardware.

### Crash Report

[hs_err_pid13608.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/13254852/hs_err_pid13608.log)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [ ] I can reproduce this crash consistently by following the exact steps I described above.
- [ ] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by Ethryan** [comment]
Have you tried using java 17 to java 19?
I don’t think Java 21 is supported yet.

**Comment by BaneWilliams** [comment]
> Have you tried using java 17 to java 19? I don’t think Java 21 is supported yet.
> 
> I’ve personally run it on 17 without any issues.

Yes, At the end of the post I tried it on J19. I haven't tried J17, but I can.

**Comment by Pxx500** [comment]
Might be AMD graphics related, I pinged you on discord help channel, you can also use !amd to get the info

**Comment by BaneWilliams** [comment]
> Might be AMD graphics related, I pinged you on discord help channel, you can also use !amd to get the info

Definitely is AMD graphics related, and reverting that driver worked (I also used DDU for the driver uninstall, but that was probably an uneccessary step).

I do like to also play latest AAA titles, so I can't keep my graphics driver here. Would you like me to do anything to further diagnose the issue?

**Comment by Pxx500** [comment]
Unfortunately it's impossible to tell if this issue will ever be fixed. Tracking it down is probably very hard and all GTNH Devs are just volunteers so it's also possible that no one wants to mess with drivers. For now just try to use the downgraded drivers and maybe in the future there will be a proper fix

**Comment by bombcar** [comment]
You could try to report a bug to AMD: https://community.amd.com/t5/drivers-software/driver-only-install-bug-reporting/td-p/587735 - no guarantee it would help but if it's reproducible they may be able to.

**Comment by FavoritoHJS** [comment]
Paging @moresteck - appears this has been happening for many months, and i fear the best option now is to assume old opengl is dead and try to port [Zink](https://docs.mesa3d.org/drivers/zink.html) over to windows...

**Comment by BaneWilliams** [comment]
Sorry to add a comment to this - just wanted to let you know that the workaround requires that the machine be offline as well as the older drivers. If you have an up to date windows, it will see a driver that old as a vulnerability and force an update.

You'll get about 8-10 hours before it does this.
